### ISSUE TITLE
`Tooltip` does not close on Escape key press

### ISSUE THREAD

**Issue by spli02** [fortanix/baklava#562]
According to ARIA Authoring Practices Guide (APG) for tooltips (https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/), a tooltip should be closed when the Escape key is pressed.
However, in the current `TooltipProvider` component, the tooltip remains open even after pressing the Escape key.

**Comment by mkrause** [comment]
There's similar discussion going on in this ticket: https://github.com/fortanix/baklava/pull/664

**Comment by nighto** [comment]
Doesn't seem to be valid anymore, please double check @mkrause

**Comment by spli02** [comment]
The issue is fixed, but the current fix seems to address the issue unintentionally, not with a proper solution.

In `ToolTipProvider.tsx`, `useComboBoxInteraction` (from `useFloatingElement.tsx`) is called with `{ enabled: false }`.
But the `useEffect` on `useComboBoxInteraction` ([L220](https://github.com/fortanix/baklava/blob/master/src/components/util/overlays/floating-ui/useFloatingElement.tsx#L220)) does not have this `enabled` check, and `onOpenChange(false, event, 'escape-key')` ([L225](https://github.com/fortanix/baklava/blob/master/src/components/util/overlays/floating-ui/useFloatingElement.tsx#L225)) is always triggered to close a floating element.

We should add the `enabled` check inside the `useEffect`, and solve the problem properly.

Suggested Change (`useFloatingElement.tsx` L220):
```
React.useEffect(() => {
  if (!enabled || !context.open) { return; }

  const watcher = new CustomCloseWatcher();

  watcher.onclose = event => {
    onOpenChange(false, event, 'escape-key');
  };

  return () => {
    watcher.destroy();
  };
}, [enabled, context.open, onOpenChange]);
```
